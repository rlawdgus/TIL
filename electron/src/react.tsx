import * as React from "react";
import * as ReactDOM from "react-dom";

const App = () => {
    return <></>;
};

ReactDOM.render(<App />, document.getElementById("app"));
