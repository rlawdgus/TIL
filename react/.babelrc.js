module.exports = {
    presets: ["@babel/env", "@babel/react", "@babel/preset-typescript"],
    plugins: ["@babel/plugin-proposal-class-properties"],
};

// @babel/preset-env : ES5+ 변환

// @babel/preset-react : React 변환

// @babel/preset-typescript : Typescript 변환
