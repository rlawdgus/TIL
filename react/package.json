{
  "name": "react",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@emotion/react": "^11.8.2",
    "@emotion/styled": "^11.8.1",
    "@reduxjs/toolkit": "^1.8.0",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@testing-library/user-event": "^13.2.1",
    "@types/jest": "^27.0.1",
    "@types/node": "^16.7.13",
    "@types/react": "^17.0.20",
    "@types/react-dom": "^17.0.9",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-redux": "^7.2.6",
    "react-scripts": "5.0.0",
    "typescript": "^4.4.2",
    "web-vitals": "^2.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "storybook": "start-storybook -p 6006 -s public",
    "build-storybook": "build-storybook -s public"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ],
    "overrides": [
      {
        "files": [
          "**/*.stories.*"
        ],
        "rules": {
          "import/no-anonymous-default-export": "off"
        }
      }
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.17.8",
    "@babel/preset-env": "^7.16.11",
    "@babel/preset-react": "^7.16.7",
    "@babel/preset-typescript": "^7.16.7",
    "@emotion/babel-plugin": "^11.7.2",
    "@storybook/addon-actions": "^6.4.20",
    "@storybook/addon-essentials": "^6.4.20",
    "@storybook/addon-interactions": "^6.4.20",
    "@storybook/addon-links": "^6.4.20",
    "@storybook/builder-webpack5": "^6.4.20",
    "@storybook/manager-webpack5": "^6.4.20",
    "@storybook/node-logger": "^6.4.20",
    "@storybook/preset-create-react-app": "^4.1.0",
    "@storybook/react": "^6.4.20",
    "@storybook/testing-library": "^0.0.9",
    "@types/redux-actions": "^2.6.2",
    "babel-loader": "^8.2.4",
    "css-loader": "^6.7.1",
    "file-loader": "^6.2.0",
    "html-webpack-plugin": "^5.5.0",
    "style-loader": "^3.3.1",
    "ts-loader": "^9.2.8",
    "webpack": "^5.71.0",
    "webpack-cli": "^4.9.2",
    "webpack-dev-server": "^4.7.4"
  }
}
