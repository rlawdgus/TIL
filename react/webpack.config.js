const path = require("path");
const HtmlWebpackPlugin = require("html-webpack-plugin");

module.exports = {
    mode: process.env.NODE_ENV || "development",
    entry: "./src/index.tsx",
    output: {
        path: path.join(__dirname, "build"),
        filename: "bundle.js",
    },
    resolve: {
        extensions: [".tsx", ".ts", ".js"],
    },
    devServer: { contentBase: path.join(__dirname, "src") },
    module: {
        rules: [
            {
                test: /\.(js|jsx)$/,
                exclude: /node_modules/,
                use: ["babel-loader"],
            },
            {
                test: /\.(ts|tsx)$/,
                exclude: /node_modules/,
                use: [
                    { loader: "ts-loader", options: { transpileOnly: true } },
                ],
            },
            {
                test: /\.(css|scss)$/,
                use: ["style-loader", "css-loader"],
            },
            {
                test: /\.(jpg|jpeg|png|gif|mp3|svg)$/,
                use: ["file-loader"],
            },
        ],
    },
    plugins: [
        new HtmlWebpackPlugin({
            template: path.join(__dirname, "public", "index.html"),
        }),
    ],
};
