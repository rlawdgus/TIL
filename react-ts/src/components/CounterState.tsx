import { useEffect, useState } from "react";

const CounterState = () => {
    const [num, setNum] = useState<number>(0);

    const increase = () => setNum(num + 1);
    const decrease = () => setNum(num - 1);

    useEffect(() => {
        console.log("CounterState componentDidUpdate");
    }, [num]);

    return (
        <>
            <h1>state</h1>
            <p>{num}</p>
            <button onClick={increase}>+1</button>
            <button onClick={decrease}>-1</button>
            {console.log("CounterState render")}
        </>
    );
};

export default CounterState;
