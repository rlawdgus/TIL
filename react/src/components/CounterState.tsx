import { useContext, useEffect, useState } from "react";

import Button from "./Button";

import { FlexContainer } from "./_styledComponent";

import { ContextExample } from "../contexts/ContextExample";

const CounterState = () => {
    const [num, setNum] = useState<number>(0);
    const numContext = useContext(ContextExample);

    const increase = () => {
        console.clear();
        setNum(num + 1);
    };
    const decrease = () => {
        console.clear();
        setNum(num - 1);
    };

    useEffect(() => {
        console.log("CounterState componentDidUpdate");
    }, [num]);

    return (
        <>
            <h1>state</h1>
            <p data-testid="state">state num: {num}</p>
            <p>context num: {numContext?.num}</p>

            <FlexContainer jus="se">
                <Button onClick={increase}>+1</Button>
                <Button onClick={decrease}>-1</Button>
            </FlexContainer>

            {console.log("CounterState render")}
        </>
    );
};

export default CounterState;
