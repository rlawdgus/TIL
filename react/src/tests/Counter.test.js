import { fireEvent, render, screen } from "@testing-library/react";

import CounterState from "../components/CounterState";

describe("counter tests", () => {
    test("CounterState test", () => {
        // eslint-disable-next-line testing-library/render-result-naming-convention
        const counter = render(<CounterState />);

        fireEvent.click(screen.getByText("+1"));

        expect(counter).toMatchSnapshot();
    });
});
