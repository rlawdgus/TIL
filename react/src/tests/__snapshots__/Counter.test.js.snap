// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterState test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        state
      </h1>
      <p>
        state num: 
        1
      </p>
      <p>
        context num: 
      </p>
      <div
        class="css-s1flh5"
      >
        <button
          class="css-1bk3ng6"
        >
          +1
        </button>
        <button
          class="css-1bk3ng6"
        >
          -1
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1>
      state
    </h1>
    <p>
      state num: 
      1
    </p>
    <p>
      context num: 
    </p>
    <div
      class="css-s1flh5"
    >
      <button
        class="css-1bk3ng6"
      >
        +1
      </button>
      <button
        class="css-1bk3ng6"
      >
        -1
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
